src/app/core/config-ui/config-field/config-field.component.ts

```diff
@@ -0,0 +1,292 @@
+      innerDataType: [this.entitySchemaField.innerDataType],
+
+      defaultValue: [this.entitySchemaField.defaultValue],
+      searchable: [this.entitySchemaField.searchable],
+      anonymize: [this.entitySchemaField.anonymize],
+      //viewComponent: [],
+      //editComponent: [],
+      //showInDetailsView: [],
+      //generateIndex: [],
+      validators: [this.entitySchemaField.validators],
+    });
+    this.form = this.fb.group({
+      id: this.fieldIdForm,
+      schemaFields: this.schemaFieldsForm,
+    });
+
+    this.updateDataTypeAdditional(this.schemaFieldsForm.get("dataType").value);
+    this.schemaFieldsForm
+      .get("dataType")
+      .valueChanges.subscribe((v) => this.updateDataTypeAdditional(v));
+    this.updateForNewOrExistingField();
+  }
+
+  private updateForNewOrExistingField() {
+    if (!!this.fieldId) {
+      // existing fields' id is readonly
+      this.fieldIdForm.disable();
+    } else {
+      const autoGenerateSubscr = this.schemaFieldsForm
+        .get("label")
+        .valueChanges.subscribe((v) => this.autoGenerateId(v));
+      // stop updating id when user manually edits
+      this.fieldIdForm.valueChanges.subscribe(() =>
+        autoGenerateSubscr.unsubscribe(),
+      );
+    }
+  }
+  private autoGenerateId(updatedLabel: string) {
+    const generatedId = generateIdFromLabel(updatedLabel);
+    this.fieldIdForm.setValue(generatedId, { emitEvent: false });
+  }
+
+  private initAvailableDatatypes(dataTypes: DefaultDatatype[]) {
+    this.dataTypes = dataTypes
+      .filter((d) => d.label !== DefaultDatatype.label) // hide "internal" technical dataTypes that did not define a human-readable label
+      .map((d) => ({
+        label: d.label,
+        value: d.dataType,
+      }));
+  }
+  objectToLabel = (v: SimpleDropdownValue) => v?.label;
+  objectToValue = (v: SimpleDropdownValue) => v?.value;
+  createNewAdditionalOption: (input: string) => SimpleDropdownValue;
+
+  private updateDataTypeAdditional(dataType: string) {
+    this.resetAdditional();
+
+    if (dataType === ConfigurableEnumDatatype.dataType) {
+      this.initAdditionalForEnum();
+    } else if (
+      dataType === EntityDatatype.dataType ||
+      dataType === EntityArrayDatatype.dataType
+    ) {
+      this.initAdditionalForEntityRef();
+    }
+
+    // hasInnerType: [ArrayDatatype.dataType].includes(d.dataType),
+
+    // TODO: this mapping of having an "additional" schema should probably become part of Datatype classes
+  }
+
+  private initAdditionalForEnum() {
+    this.typeAdditionalOptions = this.configurableEnumService
+      .listEnums()
+      .map((x) => ({
+        label: Entity.extractEntityIdFromId(x), // TODO: add human-readable label to configurable-enum entities
+        value: Entity.extractEntityIdFromId(x),
+      }));
+    this.additionalForm.addValidators(Validators.required);
+
+    this.createNewAdditionalOption = (text) => ({
```

Comment:
This at least throws an error when first used:

Add new field with new enum option -> save -> go to details and open popup for new field with new enum -> add some values -> close -> console shows errors

I guess the "no enum needs to exist" approach is not 100% working yet.